empress_gui.py

```diff
@@ -0,0 +1,335 @@
+
+    def check_dup_entry(self, *args):
+        """Checks and updates the duplication cost when user enters a value in the entry box."""
+        try:
+            self.numbers_dup_entry = float(self.dup_cost.get())
+            if self.numbers_dup_entry >= 0:
+                self.valid_dup_entry = self.numbers_dup_entry
+                self.dup_cost.set(round(self.valid_dup_entry, 2))
+                self.enable_recon_btn()
+            else:
+                messagebox.showinfo("Warning", "Please enter a non-negative number.")
+        except:
+            messagebox.showinfo("Warning", "Please enter a non-negative number.")
+    
+    def check_trans_entry(self, *args):
+        """Checks and updates the transfer cost when user enters a value in the entry box."""
+        try:
+            self.numbers_trans_entry = float(self.trans_cost.get())
+            if self.numbers_trans_entry >= 0:
+                self.valid_trans_entry = self.numbers_trans_entry
+                self.trans_cost.set(round(self.valid_trans_entry, 2))
+                self.enable_recon_btn()
+            else:
+                messagebox.showinfo("Warning", "Please enter a non-negative number.")
+        except:
+            messagebox.showinfo("Warning", "Please enter a non-negative number.")
+    
+    def check_loss_entry(self, *args):
+        """Checks and updates the loss cost when user enters a value in the entry box."""
+        try:
+            self.numbers_loss_entry = float(self.loss_cost.get())
+            if self.numbers_loss_entry >= 0:
+                self.valid_loss_entry = self.numbers_loss_entry
+                self.loss_cost.set(round(self.valid_loss_entry, 2))
+                self.enable_recon_btn()
+            else:
+                messagebox.showinfo("Warning", "Please enter a non-negative number.")
+        except:
+            messagebox.showinfo("Warning", "Please enter a non-negative number.")
+
+    def enable_recon_btn(self):
+        """Enables the next step, viewing reconciliation results, after DTL costs are set correctly."""
+        if self.valid_dup_entry is not None and self.valid_trans_entry is not None and self.valid_loss_entry is not None:
+            self.compute_recon_button.configure(state=tk.NORMAL)
+
+    def recon_analysis(self):
+        """Displays reconciliation results in numbers and further viewing options for graphical analysis."""
+        # creates a frame for the three checkbuttons
+        recon_checkbox_frame = tk.Frame(self.func_frame)
+        recon_checkbox_frame.grid(row=3, column=0, sticky="nsew")
+        recon_checkbox_frame.pack_propagate(False)
+        self.recon_space_btn_var = tk.IntVar()
+        self.recons_btn_var = tk.IntVar()
+        self.histogram_btn_var = tk.IntVar()
+        recon_space_btn = tk.Checkbutton(recon_checkbox_frame, text="View solution space", padx=10, variable=self.recon_space_btn_var, command=self.open_and_close_window_recon_space)
```

Comment:
Write this in multiple lines.
```python
recon_space_btn = tk.Checkbutton(recon_checkbox_frame, text="View solution space", 
    padx=10, variable=self.recon_space_btn_var,
    command=self.open_and_close_window_recon_space)
```
Something like this.

Reply:
Done:)